examples/packet_sculpting.rs

```diff
@@ -18,16 +20,19 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     let res_string = result[0]
         .iter()
-        .fold(String::new(), |acc, num| {
-            acc + "0x" + &num.to_string() + " "
-        })
+        .fold(String::new(), |acc, num| format!("{}{:02x}", acc, num))
```

Comment:
Won't this create a new String for every Packet? What about something like
```rs
.fold(String::new(), |acc, num|{
    acc += &format!("{:02x}", num);
    acc
})
```

Reply:
Good catch. But this is per se no worse than original implementation. Or the suggested one still suffers from the same issue. Better is to just `collect` and `join` in the end.